Import data as follows:

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns

df = sns.load_dataset("penguins")
df

,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,Male
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,Female
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,Female
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,NaN
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,Female
...,...,...,...,...,...,...,...
339,Gentoo,Biscoe,NaN,NaN,NaN,NaN,NaN
340,Gentoo,Biscoe,46.8,14.3,215.0,4850.0,Female
341,Gentoo,Biscoe,50.4,15.7,222.0,5750.0,Male
342,Gentoo,Biscoe,45.2,14.8,212.0,5200.0,Female


1,  Show Python code in the following code box which lists the number of penguins, broken down by species


In [2]:
df.species.value_counts()


species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

2.  Show Python code in the following code box which lists the number of penguins, broken down by island and sex

In [4]:
df.groupby(['island','sex']).size()


island     sex   
Biscoe     Female    80
           Male      83
Dream      Female    61
           Male      62
Torgersen  Female    24
           Male      23
dtype: int64

3.  Show Python code in the following code box which lists the mean bill length of all of the penguins.  Don't use describe().  Then use describe() and compare your results.


In [9]:
print(df.bill_length_mm.mean())


43.9219298245614


In [10]:
df.bill_length_mm.describe()

count    342.000000
mean      43.921930
std        5.459584
min       32.100000
25%       39.225000
50%       44.450000
75%       48.500000
max       59.600000
Name: bill_length_mm, dtype: float64

The mean bill length is approximately 43.92 mm using both methods

4.  Show Python code  in the following code box which prints the following statistics of the bill length of all of the penguins, broken down by the species:  minimum, median, mean and maximum


In [14]:
ss = df.groupby('species').agg({'bill_length_mm':['min','median','mean','max']})

bill_length_mm                        
                     min median       mean   max
species                                         
Adelie              32.1  38.80  38.791391  46.0
Chinstrap           40.9  49.55  48.833824  58.0
Gentoo              40.9  47.30  47.504878  59.6

5.  Repeat the above, but sort the list by the minimum bill length in ascending order.  Hint:  You may need to figure out exactly what the columns names are to sort by, and will have to do some research on how to do this.



In [25]:
ss = df.groupby('species').agg({'bill_length_mm':['min','median','mean','max']})
ss.sort_values(('bill_length_mm', 'min'), ascending=True)

                                                                                         

bill_length_mm                        
                     min median       mean   max
species                                         
Adelie              32.1  38.80  38.791391  46.0
Chinstrap           40.9  49.55  48.833824  58.0
Gentoo              40.9  47.30  47.504878  59.6

The code below is to caluclate how many missing data points there are for statistics referenced in the report


In [26]:
df['island'].isna().sum()

np.int64(0)

In [27]:
df['sex'].isna().sum()

np.int64(11)

In [35]:
df[df['sex'].isna()].groupby('island').size()

island
Biscoe       5
Dream        1
Torgersen    5
dtype: int64

In [28]:
df['bill_length_mm'].isna().sum()

np.int64(2)

In [36]:
df[df['bill_length_mm'].isna()].groupby('island').size()

island
Biscoe       1
Torgersen    1
dtype: int64